h/static/scripts/tests/controllers/share-widget-controller-test.js

```diff
@@ -0,0 +1,154 @@
+      </div>
+      <div class="share-widget-msg share-widget-msg-group">GROUP</div>
+      <div class="share-widget-msg share-widget-msg-private">PRIVATE</div>
+    </div>
+  </div>
+  `;
+
+  const template = `
+    <button id="btn1" class="js-share-widget"
+      share-widget-config='{"url":"url1", "private":false, "group":false}'>
+      btn
+    </button>
+    <button id="btn2" class="js-share-widget"
+      share-widget-config='{"url":"url2", "private":true, "group":false}'>
+      btn
+    </button>
+    <button id="btn3" class="js-share-widget"
+      share-widget-config='{"url":"url3", "private":false, "group":true}'>
+      btn
+    </button>
+    <button id="btn4" class="js-share-widget"
+      share-widget-config='{"url":"url4", "private":true, "group":true}'>
+      <span><b>btn</b></span>
+    </button>
+  `;
+
+  let container;
+
+  const getTriggers = function() {
+    return Array.from(container.querySelectorAll('.js-share-widget'));
+  };
+
+  const widgetIsVisble = function() {
+    return container.querySelector('.share-widget-template').style.visibility === 'visible';
+  };
+
+  const urlsReflectUpdate = function(addedValue) {
+    const widget = window.h.shareWidget._widget;
+    return Array.from(widget.querySelectorAll('[share-target-href]')).every((anchor) => {
+      const tmpl = anchor.getAttribute('share-target-href');
+      return tmpl.replace('{href}', addedValue) === anchor.href;
+    });
+  };
+
+  const privateMessageVisible = function() {
+    const widget = window.h.shareWidget._widget;
+    return widget.querySelector('.share-widget-msg-private').style.display === 'block';
+  };
+
+  const groupMessageVisible = function() {
+    const widget = window.h.shareWidget._widget;
+    return widget.querySelector('.share-widget-msg-group').style.display === 'block';
+  };
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    container.innerHTML = template + widgetTemplate;
+
+    document.body.appendChild(container);
+
+    getTriggers().forEach((widget) => {
+      new ShareWidgetController(widget);
+    });
+  });
+
+  afterEach(() => {
+    container.remove();
+    if (window.h) {
+      delete window.h;
+    }
+  });
+
+  it('shows/hides widget on clicking', (done) => {
+    assert.ok(window.h.shareWidget, 'Make sure we widget was initialized');
+
+    assert.isFalse(widgetIsVisble(), 'not visible by default');
+
+    const btns = getTriggers();
+
+    syn
+      .click(btns[0], () => {
+        assert.isTrue(widgetIsVisble(), 'visible after clicking trigger');
+      })
+      .click(btns[1], () => {
+        assert.isFalse(widgetIsVisble(), 'hidden after clicking another trigger while it is open');
+      })
+      .click(btns[3].querySelector('b'), () => {
+        assert.isTrue(widgetIsVisble(), 'opens when clicking elements inside of trigger');
+        done();
+      });
+  });
+
+  it('updates template properly per the configs', (done) => {
```

Comment:
Can we rephrase this in terms of how a user would see it? eg. "Share widget displays correctly for current share link". The reason being to make it more obvious what is broken if the test fails in a certain environment or as a result of refactoring.